### PR TITLE
azure-pipelines.yml: use pipeline-templates/job/code-build.yml, unquoted values and add PR builds

### PR DESCRIPTION
**Pull request by CloudPlatformer**
Edits to azure-pipelines.yml:

- Use pipeline-templates/job/code-build.yml
- Unquote values (Discussion on quotes for strings in YAML at https://stackoverflow.com/a/22235064)
- Added PR builds so DependencyCheck task runs: https://github.com/SkillsFundingAgency/das-platform-building-blocks/pull/169

### CODE REVIEW

File: pipeline-templates/job/code-build.yml [hunk 1]
```diff
@@ -0,0 +1,47 @@
+parameters:
+  SolutionBaseName:
+  BuildConfiguration:
+
+jobs:
+- job: CodeBuild
+  variables:
+  - group: BUILD Management Resources
+  steps:
+    - task: gittools.gittools.setup-gitversion-task.gitversion/setup@0
```
Comment (NickGraham101) [inline pipeline-templates/job/code-build.yml]: this name can be simplified now as done [here](https://github.com/NickGraham101/gandt-devops/blob/971ee268e467e883d130405e7612e48ea163a674/AzureDevOpsTemplates/Build/StepTemplates/dotnetcore-build.yml#L21), same for line 15

File: pipeline-templates/job/code-build.yml [hunk 2]
```diff
@@ -0,0 +1,47 @@
+parameters:
+  SolutionBaseName:
+  BuildConfiguration:
+
+jobs:
+- job: CodeBuild
+  variables:
+  - group: BUILD Management Resources
```
Comment (NickGraham101) [inline pipeline-templates/job/code-build.yml:8]: Casing of `BUILD` is inconsistent

File: azure-pipelines.yml [hunk 3]
```diff
@@ -4,21 +4,16 @@ trigger:
     include:
       - "*"
 
-pr: none
-
-workspace:
-  clean: all
```
Comment (NickGraham101) [inline azure-pipelines.yml:10]: We've removed workspace clean here but not added it in either the stage or job.  Do we not want to have workspace clean in by default?

File: pipeline-templates/job/code-build.yml (hunk 2 above)
Comment (jack-cloud-platform) [reply pipeline-templates/job/code-build.yml:8]: Simplified

Comment (CloudPlatformer) [reply pipeline-templates/job/code-build.yml:8]: The new variable group 'BUILD Management Resources' matches the convention of new variable groups for capital initial 'environment'.

I think PascalCase for job name might be OK for readability

File: azure-pipelines.yml (hunk 3 above)
Comment (CloudPlatformer) [reply azure-pipelines.yml:10]: Added

File: pipeline-templates/job/code-build.yml (hunk 2 above)
Comment (NickGraham101) [reply pipeline-templates/job/code-build.yml:8]: Agreed

**Review by NickGraham101 — APPROVED** [review]

**Pull request merged**